src/mjolnir/pbfgraphparser.cc

```diff
@@ -1785,6 +1880,24 @@ struct graph_callback : public OSMPBF::Callback {
     bss_nodes_.reset(bss_nodes);
   }
 
+  // WayCallback tag handlers
+  using TagHandler = std::function<void()>;
+  std::unordered_map<std::string, TagHandler> tag_handlers_;
+  // Tag handlers capture these fields
```

Comment:
instead of capturing `this` you should have captured `&` and not had to make all of these temporaries into member variables

Reply:
> instead of capturing `this` you should have captured `&` and not had to make all of these temporaries into member variables

The idea of making these variables be fields is to construct `tag_handlers` map once(i.e in constructor). I didn't measure but I guess constructing map on every `way_callback` call may affect performance quite a lot

Reply:
ah i see that is a good reason to do this. its unfortunate though since they are temporaries but the messiness is worth the speed up